File: app/src/main/java/com/volkanhotur/basemvvm/android/data/domain/BaseUseCase.java
package com.volkanhotur.basemvvm.android.data.domain;

import com.domain.remote.usecase.AbstractUseCase;
import com.volkanhotur.basemvvm.android.data.repository.ApiDataSource;

import java.util.HashMap;

/**
 * @author volkanhotur
 */

public abstract class BaseUseCase<T> extends AbstractUseCase<T, HashMap<String, Object>> {

    public ApiDataSource apiDataSource;
    public HashMap<String, Object> params;

    public BaseUseCase(ApiDataSource apiDataSource) {
        this.apiDataSource = apiDataSource;
        this.params = new HashMap<>();
    }

    public void addParam(String key, Object data) {
        this.params.put(key, data);
    }
}

File: app/src/main/java/com/volkanhotur/basemvvm/android/data/domain/BaseUseCase.kt
package com.volkanhotur.basemvvm.android.data.domain

import com.domain.remote.usecase.AbstractUseCase
import com.volkanhotur.basemvvm.android.data.repository.ApiDataSource
import java.util.*

/**
 * @author volkanhotur
 */
abstract class BaseUseCase<T>(var apiDataSource: ApiDataSource) : AbstractUseCase<T, HashMap<String?, Any?>?>() {

    private var params: HashMap<String, Any> = HashMap()

    fun addParam(key: String, data: Any) {
        params[key] = data
    }

}